pkg/issuer/venafi/client/request_test.go

```diff
@@ -225,9 +226,10 @@ type testSignT struct {
 	csrPEM []byte
 	client connector
 
-	expectedErr bool
+	expectedErr        bool
+	expectedRequestErr bool
```

Comment:
With this PR, we've basically removed the idea of `Sign` in favour of two different functions. I think we should therefore adjust the tests accordingly too - this set of tests is now testing *two* functions, which is going to be v confusing and doesn't lend to extending our tests in future very well.

Can we refactor this to being two different `Test` methods, which work distinct from each other? If we need to test these two 'integrated' together, I think we should test the actual code that exercises both of these codepaths (i.e. the controller itself)